lti_tool/utils.py

```diff
@@ -0,0 +1,57 @@
+from django.conf import settings
+from datetime import datetime
+from pylti1p3.grade import Grade
+from pylti1p3.contrib.django import DjangoMessageLaunch, DjangoCacheDataStorage, DjangoDbToolConf
+from pylti1p3.lineitem import LineItem
+from django.contrib.auth.models import User
+from userprofile.models import UserProfile
+from course.models import Enrollment
+
+def get_launch_url(request):
+    target_link_uri = request.POST.get('target_link_uri', request.GET.get('target_link_uri'))
+    if not target_link_uri:
+        raise Exception('Missing "target_link_uri" param')
+    return target_link_uri
+
+def get_tool_conf():
+    return DjangoDbToolConf()
+
+def get_launch_data_storage():
+    return DjangoCacheDataStorage()
+
+def send_lti_points(request, submission):
+    from exercise.exercise_summary import UserExerciseSummary
```

Comment:
This could be at the beginning of file with other imports, no?

Reply:
That caused a circular import error that I was unable to resolve otherwise - if this is undesirable, do you have any tips that could fix the error by other means?
```
  File "/srv/aplus/exercise/models.py", line 2, in <module>
    from .exercise_models import *
  File "/srv/aplus/exercise/exercise_models.py", line 46, in <module>
    from .cache.exercise import ExerciseCache
  File "/srv/aplus/exercise/cache/exercise.py", line 11, in <module>
    from ..protocol.aplus import load_exercise_page
  File "/srv/aplus/exercise/protocol/aplus.py", line 12, in <module>
    from lti_tool.utils import send_lti_points
  File "/srv/aplus/lti_tool/utils.py", line 9, in <module>
    from exercise.exercise_summary import UserExerciseSummary
  File "/srv/aplus/exercise/exercise_summary.py", line 9, in <module>
    from .cache.content import CachedContent
  File "/srv/aplus/exercise/cache/content.py", line 10, in <module>
    from ..models import LearningObject, BaseExercise
ImportError: cannot import name 'LearningObject' from partially initialized module 'exercise.models' (most likely due to a circular import) (/srv/aplus/exercise/models.py)
```